### PR TITLE
Re-enable sharing API for recent NC versions

### PR DESCRIPTION
**Pull request by DanRiess**
As discussed in #197, here is a PR that addresses the latest share API issues and re-enables sharing for recent NC versions, while keeping the ability to retrieve webapppassword tokens on all NC versions that I have tested.

### CODE REVIEW

**Comment by pbek** [comment]
@DanRiess, could you please test this?

File: docs/example/src/packages/common/assets/icons/Crop.svg
Comment (pbek) [inline docs/example/src/packages/common/assets/icons/Crop.svg:1]: What did you change here, or was this an accident?

Comment (pbek) [reply docs/example/src/packages/common/assets/icons/Crop.svg:1]: If yes, can you please remove the change from your commit. 😉

Comment (DanRiess) [reply docs/example/src/packages/common/assets/icons/Crop.svg:1]: Git was yelling at me that there were 2 files, crop.svg and Crop.svg. It made this change automatically and I couldn't revert it. So yeah, let's call it an accident

Comment (pbek) [reply docs/example/src/packages/common/assets/icons/Crop.svg:1]: Could you please remove the change and force push? 😊

Comment (DanRiess) [reply docs/example/src/packages/common/assets/icons/Crop.svg:1]: I believe it should be gone now.

Comment (pbek) [reply docs/example/src/packages/common/assets/icons/Crop.svg:1]: Do you want to squash your commits? Otherwise, I will while merging...

**Comment by pbek** [comment]
What I tested:

- [x] Test if Pull Request does not break app on latest Nextcloud 27 (27.1.11.3)
- [x] Test if Pull Request does not break app on latest Nextcloud 28 (28.0.8.1)
- [x] Test if Pull Request does not break app on latest Nextcloud 29 (29.0.4)

What's left is to test if the file sharing API works, @aleixq.

File: docs/example/src/packages/common/assets/icons/Crop.svg
Comment (DanRiess) [reply docs/example/src/packages/common/assets/icons/Crop.svg:1]: If you could do it while merging, I'd prefer that. Haven't done it before and chances are I'll break something

Comment (pbek) [reply docs/example/src/packages/common/assets/icons/Crop.svg:1]: Life got a lot easier when I started using lazygit and learned about fixups.

**Comment by aleixq** [comment]
Sorry to be late I am on vacation righ tnow. It works! using nextcloud 29.0.2

I would like to keep the class with minimal changes,  but redefining the constructor seems a way to make it work, by now... 
Because it seems like this is a short workaround solution because I am seeing changes in nc 30 shareapicontroller constructor... : 
https://github.com/nextcloud/server/blob/v30.0.0beta5/apps/files_sharing/lib/Controller/ShareAPIController.php#L73
So I suppose that there will be problems in 30... we can push now, but the list of classes uses will grow to cover versions, and I think that at some point some use class sentence will fail because of deprecation or so... we need to test for 30. Maybe a simple functional tests must be done to cover these changes.

When I get back to work in september I'll try again to claim for a native solution in nextcloud/server side... @DanReiss , your use case must be shown in https://github.com/digital-blueprint/webapppassword/issues/197#issuecomment-2216992094 !  Just to show to nc server team that allowing fine-tuning of what's exposed (such as filesharing or preview api)  will be useful.

**Comment by pbek** [comment]
Thank you for testing!

File: docs/example/src/packages/common/assets/icons/Crop.svg
Comment (pbek) [reply docs/example/src/packages/common/assets/icons/Crop.svg:1]: https://github.com/jesseduffield/lazygit changed my git-life 😉 
Pretty insightful: https://www.youtube.com/watch?v=CPLdltN7wgE&feature=youtu.be

**Pull request merged**